group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -3574,6 +3576,70 @@ public void testOnLoaded() {
 
         // foo-1 should also have a revocation timeout in place.
         assertNotNull(context.timer.timeout(consumerGroupRebalanceTimeoutKey("foo", "foo-1")));
+
+        // The classic group size counter is scheduled.
+        assertEquals(
+            GroupCoordinatorMetricsShard.DEFAULT_GROUP_GAUGES_UPDATE_INTERVAL_MS,
+            context.timer.timeout(CLASSIC_GROUP_SIZE_COUNTER_KEY).deadlineMs - context.time.milliseconds()
+        );
+    }
+
+    @Test
+    public void testScheduleClassicGroupSizeCounter() {
+        String groupId0 = "group-0";
+        String groupId1 = "group-1";
+        String groupId2 = "group-2";
+        String groupId3 = "group-3";
+        String groupId4 = "group-4";
+
+        GroupMetadataManagerTestContext context = new GroupMetadataManagerTestContext.Builder()
+            .withConsumerGroup(new ConsumerGroupBuilder(groupId0, 10))
+            .build();
+
+        ClassicGroup group1 = context.groupMetadataManager.getOrMaybeCreateClassicGroup(groupId1, true);
+        ClassicGroup group2 = context.groupMetadataManager.getOrMaybeCreateClassicGroup(groupId2, true);
+        ClassicGroup group3 = context.groupMetadataManager.getOrMaybeCreateClassicGroup(groupId3, true);
+        ClassicGroup group4 = context.groupMetadataManager.getOrMaybeCreateClassicGroup(groupId4, true);
+
+        context.groupMetadataManager.onLoaded();
+
+        // The classic group size counter is scheduled.
+        assertEquals(
+            GroupCoordinatorMetricsShard.DEFAULT_GROUP_GAUGES_UPDATE_INTERVAL_MS,
+            context.timer.timeout(CLASSIC_GROUP_SIZE_COUNTER_KEY).deadlineMs - context.time.milliseconds()
+        );
+        verify(context.metrics, times(1)).setClassicGroupGauges(argThat(
+            map -> map.get(ClassicGroupState.PREPARING_REBALANCE).get() == 0
+                && map.get(ClassicGroupState.COMPLETING_REBALANCE).get() == 0
+                && map.get(ClassicGroupState.STABLE).get() == 0
+                && map.get(ClassicGroupState.DEAD).get() == 0
+                && map.get(ClassicGroupState.EMPTY).get() == 4
+        ));
```

Comment:
nit: Couldn't we use `eq()`?

Reply:
It was because `eq()` compares `AtomicLong` by their reference. Now we use `Long` instead so we can use `eq()`